mlir/test/IR/invalid.mlir

```diff
@@ -675,3 +675,17 @@ func.func @error_at_end_of_line() {
 // -----
 
 @foo   // expected-error {{expected operation name in quotes}}
+
+// -----
+
+func.func @drop_references_on_block_parse_error(){
+  "test.user"(%i, %1) : (index, index) -> ()
+  "test.op_with_region"() ({
+  ^bb0(%i : index):
+    // expected-error @below{{expected operation name in quotes}}
```

Comment:
What is the line location for the error? Can we ensure with the diagnostic here that we check the expected line?

Reply:
That issue is unrelated to this PR. It it seems to be a lexer error. The token at which the error is produced is `123`, but its location is right before the new-line character. I'm looking into this separately.

Reply:
Actually, the location is correct. I think the way we print this is on purpose.

I found this in `Parser.cpp`:
```c++
    // Drop the \n so we emit the diagnostic at the end of the line.
    startOfBuffer = startOfBuffer.drop_back();
```

Reply:
I think this is the reason:
```mlir
// This makes sure we emit an error at the end of the correct line, the : is
// expected at the end of foo, not on the return line.
func.func @error_at_end_of_line() {
  %0 = "foo"()
  // expected-error@-1 {{expected ':' followed by operation type}}

  // This is a comment and so is the thing above.
  return
}
```
Here, it's better to report the error at the end of the line. In this example test case that I posted, the other way around would have been better...

Reply:
OK, well that's a story for another day (unless you're motivated ;) )